### PR TITLE
Store creation M2: domain selector updates

### PR DESCRIPTION
**Pull request by jaclync**
Parts of #8045 

### Description

This PR updated the domain selector screen based on the design feedback and updates in the past two weeks:

- Update placeholder image - dark mode, transparent background
- ~~Auto-focus the text field~~ --> I ended up not doing this because the search is pre-filled with the store name in the previous screen
- Update no search results error
- Update spacing between elements
- Update search field trailing padding, and decrease space around the search icon ([ref](https://github.com/woocommerce/woocommerce-ios/pull/8092#pullrequestreview-1176782138))
- Update bottom Continue CTA to use `safeAreaInset`
- Update SUGGESTIONS header text to footnote font
- Remove the skip button pecCkj-dp-p2
- Auto-dismiss the keyboard after selecting a domain and the continue button appears ([ref](https://github.com/woocommerce/woocommerce-ios/pull/8092#pullrequestreview-1176782138))

### Testing instructions

Please ignore the back button text in the navigation bar, this is updated in a separate PR https://github.com/woocommerce/woocommerce-ios/pull/8154.

- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the domain selector should be shown with a placeholder image
- Enter some store name and tap `Continue` --> the store name should be the initial query in the next domain search screen, and some domain results are loaded
- Remove the search query --> the empty state should match design and looks good in dark mode
- Enter some other search query --> the spinner should appear, and some search results are loaded shortly
- Tap to select a domain --> a `Continue` CTA should appear at the bottom, and the keyboard should be resigned
- Enter some very long query --> there should be some space in the search field trailing edge
- Replace the search query to be `Test` to have no results --> an error message should be shown, center-aligned

### Screenshots

empty query | suggestions | domain selected | loading | error
-- | -- | -- | -- | --
Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 25 44 | Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 00 | Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 04 | Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 20 | Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 15
Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 30 | Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 38 | Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 40 | Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 48 | Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 27 19

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Domains/DomainSelectorView.swift [hunk 1]
```diff
@@ -79,88 +58,112 @@ struct DomainSelectorView: View {
...
+                SearchHeader(text: $viewModel.searchTerm,
+                             placeholder: Localization.searchPlaceholder,
+                             customizations: .init(backgroundColor: .clear,
+                                                   borderColor: .separator,
+                                                   internalHorizontalPadding: 21,
+                                                   internalVerticalPadding: 12,
+                                                   iconSize: .init(width: 14, height: 14)))
+                .focused($textFieldIsFocused)
+
+                if viewModel.searchTerm.isEmpty {
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Domains/DomainSelectorView.swift]: Nit: IMO, to cover the different scenarios, we could use a State enum in the view model with cases like empty, loading, error and suggestions and use a switch case here.

**Review by selanthiraiyan — APPROVED** [review]
It works as described. 🎉 

🚢

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Domains/DomainSelectorView.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/Domains/DomainSelectorView.swift]: Thanks for your suggestion! I like this state enum approach, and updated in https://github.com/woocommerce/woocommerce-ios/pull/8165/commits/1b685f46f537ccffbc5842deb96c9ab1cf8c8acb if you'd like to take a look 🙂 (I kept the main conditional logic the same but now it's in the view model and testable)

Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/Domains/DomainSelectorView.swift]: Thanks for the update! It looks great. 🎉

**Pull request merged**